### ISSUE TITLE
Implement better optimizer class

### ISSUE THREAD

**Issue by jabalazs** [jabalazs/implicit_emotion#11]
Related to #9
This class should be able to:
* Accept specific parameters for the underlying pytorch optimizers
* Implement different learning rate schedules
  - Transformer's linear increase & O(sqrt(n)) decrease ([paper](https://papers.nips.cc/paper/7181-attention-is-all-you-need))
  - Triangular learning rates ([paper](https://arxiv.org/abs/1506.01186))
  - Slanted triangular learning rates ([paper](https://arxiv.org/abs/1801.06146))
  - Linearly decreasing learning rates
* Switch the underlying pytorch optimizer during training
